src/Network/OAuth/OAuth2/Internal.hs

```diff
@@ -63,12 +67,62 @@ instance FromJSON OAuth2Token where
 instance ToJSON OAuth2Token where
     toEncoding = genericToEncoding defaultOptions { fieldLabelModifier = camelTo2 '_' }
 
+-- | Possible OAuth error types
+-- | Not all of them can appear everywhere, but this is good enough for now.
```

Comment:
the only question I have is it makes debug a little harder since some of error could appear at both Auth Code flow and Access Token flow.
I prepend some boilerplate code into error response (HttpClient.hs) mainly helping trouble shooting. any idea to preserve such idea will be great. (my intuitive is splitting error into 2 groups)